internal/pkg/message/outbox.go

```diff
@@ -99,6 +99,11 @@ func (ob *Outbox) Send(ctx context.Context, from, to address.Address, value type
 	if err != nil {
 		return cid.Undef, nil, errors.Wrap(err, "invalid params")
 	}
+	// Our encoder can't distinguish between producing a nil and producing an empty byte array,
```

Comment:
nit: this comment is a bit confusing.  I think the situation is the opposite: our encoder does distinguish between serializing nil (as `0xf6`) and and empty byte slice (as `0x40`).

I'd prefer referencing the filecoin spec requirement here if we want a comment making the motivation clear.  i.e. "The spec's message syntax validation rules restricts empty parameters to be encoded as an empty byte string not cbor null"

Reply:
I'm saying our encoder cannot produce zero bytes as an output. If you encode an object that implements CBORMarshaler and writes zero bytes, then this encode will return a `nil`, not `[]byte{}`. This is different to encoding a `[]byte{}`, which produces one byte.

That nil is then the wrong value to put in this structure, which is itself then encoded.